### PR TITLE
feat: gc untagged manifests and signatures without references

### PR DESCRIPTION
**Pull request by peusebiu**
Signed-off-by: Petu Eusebiu <peusebiu@cisco.com>

**What type of PR is this?**

**Which issue does this PR fix**:

**What does this PR do / Why do we need it**:

**If an issue # is not available please add repro steps and logs showing the issue**:

**Testing done on this change**:

**Automation added to e2e**:

**Will this break upgrades or downgrades?**

**Does this PR introduce any user-facing change?**:

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: pkg/storage/local/local.go [hunk 1]
```diff
@@ -1477,12 +1491,131 @@ func ensureDir(dir string, log zerolog.Logger) error {
...
+	foundSubject = false
+
+	for _, notationDesc := range notationDescriptors {
+		// check if we can find the manifest which the signature points to
+		artManifest, err := getArtifactManifestByDesc(dir, notationDesc, is.log)
+		if err != nil {
+			return err
+		}
+
+		if artManifest.ArtifactType != "application/vnd.cncf.notary.v2.signature" {
```
Comment (andaaron) [inline pkg/storage/local/local.go:1597]: Isn't there a constant defined in the notation library for this?

File: pkg/storage/storage_test.go [hunk 2]
```diff
@@ -508,16 +508,16 @@ func TestStorageAPIs(t *testing.T) {
 							manifestBuf, err = json.Marshal(manifest)
 							So(err, ShouldBeNil)
 							digest := godigest.FromBytes(manifestBuf)
-							_, err = imgStore.PutImageManifest("test", digest.String(),
+							_, err = imgStore.PutImageManifest("test", "0.0.1",
```
Comment (andaaron) [inline pkg/storage/storage_test.go:511]: Why do we need these changes?

File: pkg/storage/local/local.go (hunk 1 above)
Comment (peusebiu) [reply pkg/storage/local/local.go]: fixed, thanks

File: pkg/storage/storage_test.go (hunk 2 above)
Comment (peusebiu) [reply pkg/storage/storage_test.go]: this was before adding isBlobOlderThan(gc.delay) check, thanks for the heads up, I have reverted these changes.

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**